File: 541.反转字符串-ii.py
#
# @lc app=leetcode.cn id=541 lang=python
#
# [541] 反转字符串 II
#

# @lc code=start
class Solution(object):
    def reverseStr(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        a = list(s)
        for i in xrange(0, len(a), 2*k):
            a[i:i+k] = reversed(a[i:i+k])
        return ''.join(a)
# @lc code=end

    

File: 541.反转字符串-ii.cpp
/*
 * @lc app=leetcode.cn id=541 lang=cpp
 *
 * [541] 反转字符串 II
 */

// @lc code=start
class Solution {
public:
    string reverseStr(string s, int k) {
        int n = s.size();
        string res = "";
        string tmp = "";
        for(int i = 0; i < n ; i++){
            tmp += s[i];
            if(tmp.size() / k == 2){
                reverse(tmp.begin(), tmp.begin()+k);
                res += tmp;
                tmp = "";
            }
        }
        if(tmp.size() < k){
            reverse(tmp.begin(), tmp.end());
            res += tmp;
        }
        else if(tmp.size() < 2*k and tmp.size() >= k){
            reverse(tmp.begin(), tmp.begin() + k);
            res += tmp;
        }
        return res;


    }
};
// @lc code=end

